rm-config/src/keymap.rs

```diff
@@ -0,0 +1,425 @@
+            GeneralAction::GoToEnd => "scroll to the end",
+        }
+    }
+}
+
+impl From<GeneralAction> for Action {
+    fn from(value: GeneralAction) -> Self {
+        match value {
+            GeneralAction::ShowHelp => Action::ShowHelp,
+            GeneralAction::Quit => Action::Quit,
+            GeneralAction::SoftQuit => Action::SoftQuit,
+            GeneralAction::SwitchToTorrents => Action::ChangeTab(1),
+            GeneralAction::SwitchToSearch => Action::ChangeTab(2),
+            GeneralAction::Left => Action::Left,
+            GeneralAction::Right => Action::Right,
+            GeneralAction::Down => Action::Down,
+            GeneralAction::Up => Action::Up,
+            GeneralAction::Search => Action::Search,
+            GeneralAction::SwitchFocus => Action::ChangeFocus,
+            GeneralAction::Confirm => Action::Confirm,
+            GeneralAction::Select => Action::Select,
+            GeneralAction::ScrollPageDown => Action::ScrollDownPage,
+            GeneralAction::ScrollPageUp => Action::ScrollUpPage,
+            GeneralAction::GoToBeginning => Action::Home,
+            GeneralAction::GoToEnd => Action::End,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum TorrentsAction {
+    AddMagnet,
+    Pause,
+    DeleteWithFiles,
+    DeleteWithoutFiles,
+    ShowFiles,
+    ShowStats,
+}
+
+impl UserAction for TorrentsAction {
+    fn desc(&self) -> &'static str {
+        match self {
+            TorrentsAction::AddMagnet => "add a magnet",
+            TorrentsAction::Pause => "pause/unpause",
+            TorrentsAction::DeleteWithFiles => "delete with files",
+            TorrentsAction::DeleteWithoutFiles => "delete without files",
+            TorrentsAction::ShowFiles => "show files",
+            TorrentsAction::ShowStats => "show statistics",
+        }
+    }
+}
+
+impl From<TorrentsAction> for Action {
+    fn from(value: TorrentsAction) -> Self {
+        match value {
+            TorrentsAction::AddMagnet => Action::AddMagnet,
+            TorrentsAction::Pause => Action::Pause,
+            TorrentsAction::DeleteWithFiles => Action::DeleteWithFiles,
+            TorrentsAction::DeleteWithoutFiles => Action::DeleteWithoutFiles,
+            TorrentsAction::ShowFiles => Action::ShowFiles,
+            TorrentsAction::ShowStats => Action::ShowStats,
+        }
+    }
+}
+
+#[derive(Serialize, Clone)]
+pub struct Keybinding<T: Into<Action>> {
+    pub on: KeyCode,
+    #[serde(default)]
+    pub modifier: KeyModifier,
+    pub action: T,
+}
+
+impl<T: Into<Action>> Keybinding<T> {
+    pub fn keycode_string(&self) -> String {
+        let key = match self.on {
+            KeyCode::Backspace => "Backspace".into(),
+            KeyCode::Enter => "Enter".into(),
+            KeyCode::Left => "".into(),
+            KeyCode::Right => "".into(),
+            KeyCode::Up => "".into(),
+            KeyCode::Down => "".into(),
```

Comment:
are these icons not renderable? im seeing a rectangle box in github

Reply:
These are from nerdfonts